sklearn/frozen/_frozen.py

```diff
@@ -0,0 +1,163 @@
+# Authors: The scikit-learn developers
+# SPDX-License-Identifier: BSD-3-Clause
+
+from copy import deepcopy
+
+from ..base import BaseEstimator
+from ..exceptions import NotFittedError
+from ..utils import get_tags
+from ..utils.metaestimators import available_if
+from ..utils.validation import check_is_fitted
+
+
+def _estimator_has(attr):
+    """Check that final_estimator has `attr`.
+
+    Used together with `available_if`.
+    """
+
+    def check(self):
+        # raise original `AttributeError` if `attr` does not exist
+        getattr(self.estimator, attr)
+        return True
+
+    return check
+
+
+class FrozenEstimator(BaseEstimator):
+    """Frozen estimator.
```

Comment:
I think Guillaume's suggestion helps people who don't already know what a frozen estimator is to understand it.

Some thoughts/iterations:
- "Wrap an already fitted estimator so that it can not be refit"
- "Estimator that wraps a fitted estimator to prevent refitting"